src/main/java/frc/robot/commands/RotateCommand.java

```diff
@@ -0,0 +1,62 @@
+package frc.robot.commands;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.Constants;
+import frc.robot.subsystems.DrivetrainSubsystem;
+
+public class RotateCommand extends CommandBase{
+    private DrivetrainSubsystem drive;
+    private Rotation2d target;
+    private boolean rotationDirection;
+
+    private PIDController pid = new PIDController(3, 0.5, 0.5);
+
+    public RotateCommand(DrivetrainSubsystem drive, Rotation2d target) {
+        this.drive = drive;
+        this.target = target;
+        addRequirements(drive);
+    }
+
+    @Override 
+    public void initialize() {
+        Rotation2d diff = drive.getGyroscopeRotation().minus(target);
+        if(diff.getDegrees() < 0) {
+            rotationDirection = true;
+        }else {
+            rotationDirection = false;
+        }
+    }
+
+    @Override
+    public boolean isFinished() {
+        return withinAcceptableOffset();
+    }
+
+    private boolean withinAcceptableOffset() {
+        if(drive.getGyroscopeRotation().equals(target)) {
+            return true;
+        }
+
+        if(Math.abs(drive.getGyroscopeRotation().minus(target).getDegrees()) <= Constants.ACCEPTABLE_AUTOROTATE_ERROR_DEGREES) {
+            return true;
+        } 
+
+        return false;
+    }
+
+    @Override
+    public void execute() {
+        SmartDashboard.putNumber("radians away", drive.getGyroscopeRotation().minus(target).getRadians());
+        drive.rotate(rotationDirection, pid.calculate(Math.abs(drive.getGyroscopeRotation().minus(target).getRadians())) * -1);
```

Comment:
Ideally we should not need to know the direction and the rotation direction is based on the sign of the target error. Therefore no need to take abs of the value as the sign represents the direction